cmd/kube-proxy/app/server_others.go

```diff
@@ -281,6 +282,61 @@ func (s *ProxyServer) createProxier(config *proxyconfigapi.KubeProxyConfiguratio
 				kernelHandler,
 			)
 		}
+		if err != nil {
+			return nil, fmt.Errorf("unable to create proxier: %v", err)
+		}
+	} else if config.Mode == proxyconfigapi.ProxyModeNFTables {
```

Comment:
has `platformCheckSupported()` checks the iptables requirements and is called from 
https://github.com/kubernetes/kubernetes/blob/369692ab7a037af8adac163246f46a0b1a087e07/cmd/kube-proxy/app/server.go#L641

now it has to depend on the proxyMode , because in a system that uses nftables mode with only nft and not iptables usersapce it will fail, despite it should work.

If you want to fix it later in a follow up then just add a TODO to not forget

Reply:
Added a new commit `Distinguish iptables-based and nftables-based backends, do startup cleanup` which fixes this and also adds the cleanup when switching between iptables-based and nftables-based modes.